server/clip_server/model/openclip_model.py

```diff
@@ -20,17 +28,139 @@ class OpenCLIPModel(CLIPModel):
+
+                    # patch dtype to float32 on CPU
+                    if device == "cpu":
+                        float_holder = torch.jit.trace(
+                            lambda: torch.ones([]).float(), example_inputs=[]
+                        )
+                        float_input = list(
+                            float_holder.graph.findNode("aten::to").inputs()
+                        )[1]
+                        float_node = float_input.node()
+
+                        def patch_float(module):
+                            try:
+                                graphs = (
+                                    [module.graph] if hasattr(module, "graph") else []
+                                )
+                            except RuntimeError:
+                                graphs = []
+
+                            if hasattr(module, "forward1"):
+                                graphs.append(module.forward1.graph)
+
+                            for graph in graphs:
+                                for node in graph.findAllNodes("aten::to"):
+                                    inputs = list(node.inputs())
+                                    for i in [
+                                        1,
+                                        2,
+                                    ]:  # dtype can be the second or third argument to aten::to()
+                                        if inputs[i].node()["value"] == 5:
+                                            inputs[i].node().copyAttributes(float_node)
+
+                        model.apply(patch_float)
+                        patch_float(model.encode_image)
+                        patch_float(model.encode_text)
+                        model.float()
+
+                    # ensure image_size attr available at consistent location for both jit and non-jit
+                    model.visual.image_size = model.input_resolution.item()
+                if precision == "fp32":
+                    model = model.float()
+            else:
+                if model_name in _MODEL_CONFIGS:
+                    model_cfg = deepcopy(_MODEL_CONFIGS[model_name])
+                else:
+                    raise RuntimeError(f'Model config for {model_name} not found.')
+
+                model = CLIP(**model_cfg)
+
+                if pretrained:
+                    if model_path:
+                        model.load_state_dict(load_state_dict(model_path))
+                    else:
+                        raise RuntimeError(
+                            f'Pretrained weights ({pretrained}) not found for model {model_name}.'
+                        )
+
+                model.to(device=torch.device(device))
+                if precision == "fp16":
+                    convert_weights_to_fp16(model)
+                if jit:
+                    model = torch.jit.script(model)
+
+            self._model = model
             self._model_name = model_name
+        else:
```

Comment:
we should make sure all of the models are uploaded to our s3 bucket.

Reply:
i uploaded pt models, should be available in any minutes

Reply:
update: any hours 😅poor internet